pkg/kn/commands/trigger/describe.go

```diff
@@ -123,4 +124,50 @@ func writeTrigger(dw printers.PrefixWriter, trigger *v1beta1.Trigger, printDetai
 			subWriter.WriteAttribute(key, value)
 		}
 	}
+	if len(trigger.Spec.Filters) > 0 {
+		// Split 'Filter' and 'Filters (experimental)' with new line
+		dw.WriteLine()
+		subWriter := dw.WriteAttribute("Filters (experimental)", "")
+		for _, filter := range trigger.Spec.Filters {
+			writeNestedFilters(subWriter, filter)
+		}
+	}
+}
+
+// writeNestedFilters goes through SubscriptionsAPIFilter and writes its content accordingly
+func writeNestedFilters(dw printers.PrefixWriter, filter v1beta1.SubscriptionsAPIFilter) {
+	v := reflect.ValueOf(filter)
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Type().Field(i)
+		fieldValue := v.Field(i)
+
+		// Write if it's non-zero string, fields: CESQL
+		if fieldValue.Kind() == reflect.String && !fieldValue.IsZero() {
+			dw.WriteAttribute(field.Name, fieldValue.String())
+		}
+		// Write map[string]string key:value pairs of field: Exact, Prefix, Suffix
+		if fieldValue.Kind() == reflect.Map && fieldValue.Len() > 0 {
+			for k, v := range fieldValue.Interface().(map[string]string) {
+				dw.WriteAttribute(k, v)
+			}
+		}
+
+		// iterate through []SubscriptionsAPIFilter of fields: All, Any
+		if fieldValue.Kind() == reflect.Slice {
+			for j := 0; j < fieldValue.Len(); j++ {
+				element := fieldValue.Index(j)
+				// Write filter field name only and create next indentation
+				dw = dw.WriteAttribute(field.Name, "")
+				// Call write recursively for struct SubscriptionsAPIFilter
+				if element.Kind() == reflect.Struct {
+					writeNestedFilters(dw, element.Interface().(v1beta1.SubscriptionsAPIFilter))
+				}
+			}
+		}
+
+		// Call write recursively for struct SubscriptionsAPIFilter of field: Not
+		if fieldValue.Kind() == reflect.Struct {
+			writeNestedFilters(dw, fieldValue.Interface().(v1beta1.SubscriptionsAPIFilter))
+		}
+	}
 }
```

Comment:
you have the best context here whether reflection is the best way to go or not

Reply:
My intuition was that reflection will make it a bit shorter and cleaner than hardcoded fields. Plus hopefully it will be more future proof any of the fields is added/changed. 
At the end it got a bit out of hand and it's so much shorter etc., but the benefit of having it a bit more dynamic hopefully stayed.

Reply:
That would be also my question, whether it wouldn't be easier to use the fields directly and make it easier to comprehend. When new fields come up, one probably needs to check whether they fit in the layout anyway. Not sure how it looks like, sorry, I'm a bit disconnected. But let me ask GPT-4 ...

Reply:
@rhuss here's the struct to show: https://github.com/knative/eventing/blob/main/pkg/apis/eventing/v1/trigger_types.go#L124

Reply:
> Can you tell me what the new method does and can give me a sample output what it would print ?

![image](https://user-images.githubusercontent.com/99080/227235164-40633931-2a62-49e7-b3a6-565d5264a254.png)
![image](https://user-images.githubusercontent.com/99080/227235274-dc5242d5-0ac1-49b6-83f7-dee94e2aa450.png)

Reply:
This version doesn't entirely cut it, but I think I would prefer the simpler version that is more hardcoded as the reflect-based version, while flexible, is probably a bit over the top as I think one would need to adjust the output anyway if the filter gets more complex.

Reply:
I've pushed revamped version with exact fields + unit test. https://github.com/knative/client/pull/1794/commits/0e2e46154ab35b24ff163ebd5c4f2c94f0c82124